pkg/util/pem.go

```diff
@@ -120,3 +125,19 @@ func DecodeX509CertificateChainBytes(certBytes []byte) ([]*x509.Certificate, err
 
 	return certs, nil
 }
+
+// Check if the given certificate was added to certificates bundle already
+func IsCertificateDuplicate(certsHash map[[32]byte]struct{}, certData []byte) bool {
```

Comment:
What do you think about creating a type for this function? Maybe it could be named `certPool`?  😉 With an `appendCertFromPEM` func, the type could also hold the certificates returned from the outer function.

Reply:
I added `CertPool` type type with related methods.
It's not possible to put that structure into `internal` directory, due to limitation of importing packages which are located under `internal` directory.